test/test_sparse.py

```diff
@@ -143,6 +143,79 @@ def f5():
             # Check warn-once:
             self.assertEqual(len(cm.warnings), 1)
 
+    def test_gradcheck_check_sparse_nnz(self):
+        """Tests for deprecated check_sparse_nnz keyword argument of gradcheck.
+
+        Suggested deprecation steps:
+        2.1: Specification of check_sparse_nnz triggers a warning.
+        2.2: Specification of check_sparse_nnz triggers a warning and
+             an exception. Update gradcheck and this test accordingly.
+        2.3: Specification of check_sparse_nnz triggers an
+             exception. Remove all check_sparse_nnz usages from
+             gradcheck and delete this test.
+
+        """
+        def fn(x, masked_grad):
+            return x.to_dense(masked_grad=masked_grad)
+
+        def test(x, masked_grad, masked, check_sparse_nnz):
+            x = x.detach().clone().requires_grad_()
+            r = torch.autograd.gradcheck(fn, (x, masked_grad), masked=masked, check_sparse_nnz=check_sparse_nnz)
+            self.assertTrue(r)
```

Comment:
no need to assign the output of gradcheck and do this assert since the behavior by default is to raise an exception